test/headless/specs/watcher/impression-complete-event.spec.js

```diff
@@ -44,54 +95,67 @@ testModule('Impression Complete event', class extends WatcherTestClass {
   }
 
   ['@test should fire only once when item is moved into viewport, remains for threshold time, then moves out of viewport']() {
-    return this.context.scrollTo(200)
-      .wait(IMPRESSION_THRESHOLD + RAF_THRESHOLD * 4)
+    return this.setupTest()
+      .onDOMReady()
+      .scrollTo(200)
+      .waitForNthElemEvent('fifth', 'impressed', '1')
       .scrollTo(0)
-      .wait(RAF_THRESHOLD * 3)
+      .waitForImpression(1)
```

Comment:
This test needs to specify and test that the threshold time is respected. So there has to be some sort of timer here. If I understand the test, it would still pass if the `impression-complete` event happened way early or way late.